storage/internal/apiv1/storage_client.go

```diff
@@ -0,0 +1,2388 @@
+func (c *Client) InsertObjectAccessControl(ctx context.Context, req *storagepb.InsertObjectAccessControlRequest, opts ...gax.CallOption) (*storagepb.ObjectAccessControl, error) {
+	return c.internalClient.InsertObjectAccessControl(ctx, req, opts...)
+}
+
+// ListObjectAccessControls retrieves ACL entries on the specified object.
+func (c *Client) ListObjectAccessControls(ctx context.Context, req *storagepb.ListObjectAccessControlsRequest, opts ...gax.CallOption) (*storagepb.ListObjectAccessControlsResponse, error) {
+	return c.internalClient.ListObjectAccessControls(ctx, req, opts...)
+}
+
+// PatchObjectAccessControl patches an ACL entry on the specified object.  Patch is similar to update,
+// but only applies or appends the specified fields in the
+// object_access_control object.  Other fields are unaffected.
+func (c *Client) PatchObjectAccessControl(ctx context.Context, req *storagepb.PatchObjectAccessControlRequest, opts ...gax.CallOption) (*storagepb.ObjectAccessControl, error) {
+	return c.internalClient.PatchObjectAccessControl(ctx, req, opts...)
+}
+
+// UpdateObjectAccessControl updates an ACL entry on the specified object.
+func (c *Client) UpdateObjectAccessControl(ctx context.Context, req *storagepb.UpdateObjectAccessControlRequest, opts ...gax.CallOption) (*storagepb.ObjectAccessControl, error) {
+	return c.internalClient.UpdateObjectAccessControl(ctx, req, opts...)
+}
+
+// ComposeObject concatenates a list of existing objects into a new object in the same
+// bucket.
+func (c *Client) ComposeObject(ctx context.Context, req *storagepb.ComposeObjectRequest, opts ...gax.CallOption) (*storagepb.Object, error) {
+	return c.internalClient.ComposeObject(ctx, req, opts...)
+}
+
+// CopyObject copies a source object to a destination object. Optionally overrides
+// metadata.
+func (c *Client) CopyObject(ctx context.Context, req *storagepb.CopyObjectRequest, opts ...gax.CallOption) (*storagepb.Object, error) {
+	return c.internalClient.CopyObject(ctx, req, opts...)
+}
+
+// DeleteObject deletes an object and its metadata. Deletions are permanent if versioning
+// is not enabled for the bucket, or if the generation parameter
+// is used.
+func (c *Client) DeleteObject(ctx context.Context, req *storagepb.DeleteObjectRequest, opts ...gax.CallOption) error {
+	return c.internalClient.DeleteObject(ctx, req, opts...)
+}
+
+// GetObject retrieves an object’s metadata.
+func (c *Client) GetObject(ctx context.Context, req *storagepb.GetObjectRequest, opts ...gax.CallOption) (*storagepb.Object, error) {
+	return c.internalClient.GetObject(ctx, req, opts...)
+}
+
+// GetObjectMedia reads an object’s data.
+func (c *Client) GetObjectMedia(ctx context.Context, req *storagepb.GetObjectMediaRequest, opts ...gax.CallOption) (storagepb.Storage_GetObjectMediaClient, error) {
+	return c.internalClient.GetObjectMedia(ctx, req, opts...)
+}
+
+// InsertObject stores a new object and metadata.
```

Comment:
Nice detailed docs here, has this been updated at all based on your investigations into resumable uploads @noahdietz ?

Reply:
> has this been updated at all based on your investigations into resumable uploads

Nope, this is actually just the documentation from the [proto itself](https://github.com/googleapis/googleapis/blob/master/google/storage/v1/storage.proto#L226-L250)! I agree, its quite solid.